File: core/src/com/maattss/intro/exercise/sprites/MoveButton.java
package com.maattss.intro.exercise.sprites;

import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.math.Rectangle;
import com.badlogic.gdx.math.Vector2;

public class MoveButton {
    private Texture button;
    private Vector2 position;
    private Rectangle bounds;

    public MoveButton(int x, int y, String texturePath) {
        button = new Texture(texturePath);
        position = new Vector2(x,y);
        bounds = new Rectangle(position.x, position.y,100,100);
    }

    public Rectangle getBounds() { return bounds; }

    public float getX() { return position.x; }

    public float getY() { return position.y; }

    public Texture getTexture() {
        return button;
    }

    public void dispose() { button.dispose(); }
}

File: core/src/com/maattss/intro/exercise/sprites/MoveButton.kt
package com.maattss.intro.exercise.sprites

import com.badlogic.gdx.graphics.Texture
import com.badlogic.gdx.math.Rectangle
import com.badlogic.gdx.math.Vector2

class MoveButton(x: Int, y: Int, texturePath: String?) {
    val texture: Texture
    private val position: Vector2
    val bounds: Rectangle

    val x: Float
        get() = position.x

    val y: Float
        get() = position.y

    fun dispose() {
        texture.dispose()
    }

    init {
        texture = Texture(texturePath)
        position = Vector2(x.toFloat(), y.toFloat())
        bounds = Rectangle(position.x, position.y, 100f, 100f)
    }
}